mesonbuild/modules/python.py

```diff
@@ -62,9 +63,11 @@ def __init__(self, python_holder, environment, kwargs):
         if DependencyMethods.PKGCONFIG in self.methods and not python_holder.is_pypy:
             pkg_version = self.variables.get('LDVERSION') or self.version
             pkg_libdir = self.variables.get('LIBPC')
+            pkg_embed = '-embed' if self.embed else ''
+            pkg_name = 'python-{}{}'.format(pkg_version, pkg_embed)
```

Comment:
We cannot *only* check for `python-{version}-embed.pc`, because it does not exist on python <= 3.7. The usual recommendation is to first check for -embed and fall back on the dependency without embed, but since we already know the version due to finding it from the detected python3 executable, we can conditionalize this on the version and do a single lookup.

Reply:
Thanks, added the version check.